src/cli/functions/Battle.java

```diff
@@ -11,9 +15,39 @@ public Battle() {
 
     @Override
     public void call(Game game) {
-        // TODO: implement call in Battle
-        // Must call game.findOpponent() until user is satisfied.
-        // Then call game.battle() and print battle records
-        throw new NotImplementedException();
+        Player opponent = new Player("GeraltofRivia", "whitewolf"); //for testing (GeraltofRivia should be default opponent)
+        game.setcurrentOpponent(opponent);
+        String BattleMenu = """
+                0.Find new Opponent
+                1.Start Battle
+                2.Exit""";
+
+        int command = 0;
+        while(command == 0){
+            System.out.println("Your Opponent : "+ opponent.getUsername() + "   Xp : "+ opponent.getXp());
+            System.out.println(BattleMenu);
+            Scanner scanner = new Scanner(System.in);
+            System.out.print("Enter command number: ");
+            command = scanner.nextInt();
+
+            switch (command) {
+                case 0:
+                    game.clearOpponent();
+                    opponent = game.findOpponent();
+                    game.setcurrentOpponent(opponent);
+                    if (opponent == null) {
+                        System.out.println("No available opponents. Trying again...");
+                    }
+                    break;
+                case 1:
+                    // battle
+                    break;
+                case 2:
+                    // Exit
+                    break;
+                default:
+                    System.out.println("Invalid command number");
+            }
+        }
```

Comment:
Does this have to be a menu?

expected interaction:
```text
select battle command
shows a random opponent
do you want to skip? yes
shows another random opponent
do you want to skip? no
displays the battle result/log
```
We will create the NPC in the Cli.begin if the loaded players array is empty.

```suggestion
        // TODO: implement call in Battle
        // Must call game.findOpponent() until user is satisfied.
        // Then call game.battle() and print battle records
        throw new NotImplementedException();
```